src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -1,19 +1,66 @@
 package core.basesyntax.impl;
 
+import core.basesyntax.Pair;
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    public static final int STORAGE_SIZE = 10;
+    private final Pair<K, V>[] storage;
+
+    public StorageImpl() {
+        storage = new Pair[STORAGE_SIZE];
+    }
+
     @Override
     public void put(K key, V value) {
+        Pair<K, V> toSave = new Pair<>(key, value);
+        int position = findExistingPositionByKey(key);
+        if (position != -1) {
+            storage[position] = toSave;
+        } else {
+            position = findFreeSpace();
+            if (position != -1) {
+                storage[position] = toSave;
+            } else {
+                throw new ArrayStoreException("Storage is out of free space!");
+            }
+        }
     }
 
     @Override
     public V get(K key) {
+        int position = findExistingPositionByKey(key);
+        if (position != -1) {
```

Comment:
-1 is the magic number. Use the constant from the comment above.
```suggestion
        if (position != -1) {
```